### PR TITLE
feat: add item slot in Pagination

### PR DESCRIPTION
**Pull request by jaulz**
This PR adds a slot for `item` in `Pagination`. Please let me know if you would consider this PR ~so I can also extend the docs and tests~.

### CODE REVIEW

File: src/pagination/src/interface.ts [hunk 1]
```diff
@@ -16,3 +16,12 @@ export type PaginationSizeOption = SelectBaseOption<number, string>
...
 export type RenderPrev = RenderPrefix
+
+export type RenderItem = (info: {
+  item: {
+    label: string | number
+    type: 'page' | 'fastBackward' | 'fastForward'
+  }
+  showFastBackward: boolean
+  showFastForward: boolean
+}) => VNodeChild
```
Comment (07akioni) [inline src/pagination/src/interface.ts:27]:
```
(info: {
  label: string,
  type: 'page' | 'fastBackward' | 'fastForward',
  active: boolean // hover for fastBackward & fastForward, current page for page
}) => VNodeChild
```

**Review by 07akioni — COMMENTED** [review]
I thinks `render-label` is better than `render-item` since it does not render the whole item.

File: CHANGELOG.en-US.md [hunk 2]
```diff
@@ -18,6 +18,7 @@
 - `n-data-table` `TableColumn` supports `string` typed `width`, closes [#2102](https://github.com/TuSimple/naive-ui/issues/2102).
 - `n-calendar` adds `on-panel-change` prop, closes [#2082](https://github.com/TuSimple/naive-ui/issues/2082).
 - `n-upload` adds `on-error` prop.
+- `n-pagination` adds `item` slot.
```
Comment (07akioni) [inline CHANGELOG.en-US.md:21]:
```suggestion
- `n-pagination` adds `label` slot.
```

File: CHANGELOG.zh-CN.md [hunk 3]
```diff
@@ -18,6 +18,7 @@
 - `n-data-table` `TableColumn` 支持 `string` 类型的 `width`，关闭 [#2102](https://github.com/TuSimple/naive-ui/issues/2102)
 - `n-calendar` 新增 `on-panel-change` 属性，关闭 [#2082](https://github.com/TuSimple/naive-ui/issues/2082)
 - `n-upload` 新增 `on-error` 属性
+- `n-pagination` adds `item` slot.
```
Comment (07akioni) [inline CHANGELOG.zh-CN.md:21]:
```suggestion
- `n-pagination` adds `label` slot.
```

File: CHANGELOG.zh-CN.md [hunk 4]
```diff
@@ -18,6 +18,7 @@
 - `n-data-table` `TableColumn` 支持 `string` 类型的 `width`，关闭 [#2102](https://github.com/TuSimple/naive-ui/issues/2102)
 - `n-calendar` 新增 `on-panel-change` 属性，关闭 [#2082](https://github.com/TuSimple/naive-ui/issues/2082)
 - `n-upload` 新增 `on-error` 属性
+- `n-pagination` adds `label` slot.
```
Comment (07akioni) [inline CHANGELOG.zh-CN.md:21]:
```suggestion
- `n-pagination`  `label` slot
```

File: src/pagination/src/interface.ts (hunk 1 above)
Comment (07akioni) [reply src/pagination/src/interface.ts]: If type if 'fastBackward' & 'fastForward', active means hovered. If type is 'page', active means it's current page.

Comment (07akioni) [reply src/pagination/src/interface.ts]: @jaulz

Comment (07akioni) [reply src/pagination/src/interface.ts]: The original showFastBackward & showFastForward is useless for another case and 'page' type.

Comment (jaulz) [reply src/pagination/src/interface.ts]: Yep, it's added now. Sorry I missed that before 😊

**Pull request merged**